### PR TITLE
Quarkiverse extension template improvements

### LINKED ISSUE #22034: Quarkiverse extension template improvements
## General

* When I created JGit, it somehow was replaced by Jgit (no capital G) in the extension name and several other places. I'm pretty sure I did it right and I think I had the same issue with JSch.

## Runtime

* Maybe we should have a `<description>Put a description here</description>` in the runtime template? Not worse than having the Quarkiverse parent description appearing IMHO. But I wonder if we should make a description field mandatory when creating an extension. It could then be wired automatically. We need one anyway so why not make it mandatory?
* Maybe we should point to the Quarkiverse doc by default and have a guide entry?

## Docs

https://github.com/quarkiverse/quarkus-neo4j/ is a good example for that

* docs should have a proper pom.xml, see the template for the new extension I created
  * pom also handles the copy of the potential generated config file to the right place, we probably won't get it right always given it depends on the config root but we could try using the extension name as an approximation
* docs/templates directory should be included
  * the Maven version replaced with what we have in the build)
  * the `:quarkus-neo4j-version:` would probably be better with a :project-version: to be more generic and avoid having to update it
* docs/modules/ROOT/assets/images should be included as it's not easy to figure out how to make images work
* in docs/antora.yml , `title` should be `Quarkus Extension` and not `Quarkus - Extension` (no dash). The idea is that we don't want to convey the idea it's part of Quarkus (also it messes up the order in Antora)
* docs/modules/ROOT/pages/config.adoc containing Freemarker config  should go away
* index.adoc should be updated to:
  * use the attributes.adoc we generate (see Neo4j for example)
  * use the generated config include file we generate (and copy with the pom) - maybe with an optional by default so that user can build things 

## CI

* Release workflow should be updated to take into account the updated file (see neo4j one for instance) - the block Maven Release needs the additional commit and the new block to adjust the tag needs to be in
* Quarkus snapshot workflow should be updated to point to `quarkiverse-extension` not `quarkus-extension` as the ecosystem path
* Add -Dnative in .github/workflows/build.yml - at least I did it for all the moved extensions as I wanted them to be covered

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by gsmet**
Fixes #22034

I think I have run all the required tests and updated the snapshots but who knows :).

### CODE REVIEW

**Comment by gsmet** [comment]
Hmmm, wait I want to adjust something.

**Comment by gsmet** [comment]
@ia3andy can you confirm me that we can break backward compatibility of templates as long as it's the base codestarts being adjusted? I think I remember you saying that?

**Comment by ia3andy** [comment]
> @ia3andy can you confirm me that we can break backward compatibility of templates as long as it's the base codestarts being adjusted? I think I remember you saying that?

Yes ;-)

I will have a closer look later I am AFK

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus-extension/code/extension-base/java/runtime/src/main/resources/META-INF/quarkus-extension.tpl.qute.yaml [hunk 1]
```diff
@@ -1,9 +1,9 @@
 name: {extension.name}
-#description: {extension.name} ...
+#description: {extension.description ?: 'Do something useful.'}
 metadata:
 #  keywords:
 #    - {extension.id}
-#  guide: ...
+#  guide: {extension.guide ?: '...'}
```
Comment (ia3andy) [inline independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus-extension/code/extension-base/java/runtime/src/main/resources/META-INF/quarkus-extension.tpl.qute.yaml:6]: No CLI option for this? later?

**Review by ia3andy — APPROVED** [review]
Thanks for this! It's painful to do but will be awesome for new extensions..

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus-extension/code/extension-base/java/runtime/src/main/resources/META-INF/quarkus-extension.tpl.qute.yaml (hunk 1 above)
Comment (gsmet) [reply independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus-extension/code/extension-base/java/runtime/src/main/resources/META-INF/quarkus-extension.tpl.qute.yaml:6]: Well, I don't think people will actually know the URL when creating the extension. So I just generate one that is consistent with our conventions for the Quarkiverse.

**Comment by gsmet** [comment]
Thanks for the review. Let's wait for George's review, it's a public holiday for him today.

**Review by gastaldi — APPROVED** [review]
Nice! 

Can we also change the `project.yml` for releases to have double quotes in their values (I've seen people having trouble when releasing `0.1.0` without double quotes) and also have the `next-version` as `"999-SNAPSHOT"` by default or should we do that in a separate PR?

**Comment by gsmet** [comment]
@gastaldi let me do it.

**Review by ia3andy — APPROVED** [review]
LGTM

**Review by gastaldi — APPROVED** [review]
Looks good to me. Good job!

**Pull request merged**